tools/nnyeah/nnyeah/ConstructorTransforms.cs

```diff
@@ -48,15 +54,38 @@ static bool IsIntPtrCtor (MethodDefinition d)
             return isSingle || isDouble;
         }
 
-        static bool IsNSObjectDerived (TypeReference? typeReference)
+        bool IsNSObjectDerived (TypeReference? typeReference)
         {
-            if (typeReference is null) {
+            if (typeReference is null)
```

Comment:
If you do the following it will remove all the added ? and ! in the following up code:
```csharp
if (typeReference is TypeReference initialTypeReference) {
    // you implementation foes here
}
return false;
```

That will remove simplify a few things in the following code.

Reply:
How so? There are no `!` or `?` on `typeReference` in the following code and the the call to `BaseType` can return null...